src/components/SQFormDialog/SQFormDialogInner.js

```diff
@@ -40,54 +40,94 @@ function SQFormDialogInner({
+    {openDialog: openDialogAlertModal, closeDialog: closeDialogAlertModal}
+  ] = useDialog();
+
+  const handleCancel = () => {
+    if (!openCancelConfirmationDialog) {
+      resetForm();
+      onClose();
+    } else {
+      openDialogAlertModal();
+    }
+  };
+
+  const confirmCancel = () => {
     resetForm();
     onClose();
+    closeDialogAlertModal();
   };
 
+  const {content, ...restProps} = cancelConfirmDialogProps;
+
   return (
-    <Dialog
-      disableBackdropClick={disableBackdropClick}
-      maxWidth={maxWidth}
-      open={isOpen}
-      TransitionComponent={Transition}
-      onClose={handleClose}
-    >
-      <Form>
-        <DialogTitle disableTypography={true}>
-          <Typography variant="h4">{title}</Typography>
-        </DialogTitle>
-        <DialogContent dividers={true}>
-          <Grid {...muiGridProps} container spacing={muiGridProps.spacing || 2}>
-            {children}
-          </Grid>
-        </DialogContent>
-        <DialogActions classes={actionsClasses}>
-          <RoundedButton
-            title={cancelButtonText}
-            onClick={handleClose}
-            color="secondary"
-            variant="outlined"
-          >
-            {cancelButtonText}
-          </RoundedButton>
-          {onSave && (
-            <SQFormButton
-              title={saveButtonText}
-              isDisabled={isDisabled}
-              shouldRequireFieldUpdates={shouldRequireFieldUpdates}
+    <>
+      <Dialog
+        disableBackdropClick={disableBackdropClick}
+        maxWidth={maxWidth}
+        open={isOpen}
+        TransitionComponent={Transition}
+        onClose={handleCancel}
+      >
+        <Form>
+          <DialogTitle disableTypography={true}>
+            <Typography variant="h4">{title}</Typography>
+          </DialogTitle>
+          <DialogContent dividers={true}>
+            <Grid
+              {...muiGridProps}
+              container
+              spacing={muiGridProps.spacing || 2}
+            >
+              {children}
+            </Grid>
+          </DialogContent>
+          <DialogActions classes={actionsClasses}>
+            <RoundedButton
+              title={cancelButtonText}
+              onClick={handleCancel}
+              color="secondary"
+              variant="outlined"
             >
-              {saveButtonText}
-            </SQFormButton>
-          )}
-        </DialogActions>
-      </Form>
-    </Dialog>
+              {cancelButtonText}
+            </RoundedButton>
+            {onSave && (
+              <SQFormButton
+                title={saveButtonText}
+                isDisabled={isDisabled}
+                shouldRequireFieldUpdates={shouldRequireFieldUpdates}
+              >
+                {saveButtonText}
+              </SQFormButton>
+            )}
+          </DialogActions>
+        </Form>
+      </Dialog>
+
+      {openCancelConfirmationDialog && (
+        <DialogAlert
+          isOpen={isDialogAlertModalOpen}
```

Comment:
[blocker] It looks like the only props from `cancelConfirmDialogProps` that are being used is the content.  If we are not going to use the other props we should remove them from the propTypes and possibly simplify the `cancelConfirmDialogProps` to just one prop that is a string.  Possibly `cancelConfirmDialogBody`

Reply:
If UX did not request the dialog box to be customizable we should not allow consumers to customize it.  We should just use the values that UX gave us.

Reply:
@Chris-Boyle I have updated the PR please have a look